################################################################################
SHORT DESCRIPTION: 
################################################################################
Backport of an upstream fix for the txt-stat tool.

################################################################################
LONG DESCRIPTION: 
################################################################################
Pull this into 1.8.3 because without it txt-stat segfaults from time to time.

http://hg.code.sf.net/p/tboot/code/rev/9f937d110c99

################################################################################
CHANGELOG 
################################################################################
Created:
Ross Philipson, philipsonr@ainfosec.com, 9/22/2015

################################################################################
REMOVAL 
################################################################################
It is a back port, it should be removed on the next move to a newer version.

################################################################################
UPSTREAM PLAN 
################################################################################

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None

################################################################################
PATCHES 
################################################################################
Index: tboot-1.8.3/utils/txt-stat.c
===================================================================
--- tboot-1.8.3.orig/utils/txt-stat.c	2015-09-22 13:10:40.664532203 -0400
+++ tboot-1.8.3/utils/txt-stat.c	2015-09-22 13:11:35.787394541 -0400
@@ -235,8 +235,8 @@
     printf("\t buf:\n");
     /* log->buf is phys addr of buf, which will not match where mmap has */
     /* map'ed us, but since it is always just past end of struct, use that */
-    /* Uncompress tboot log */ 
-    LZ_Uncompress(log_buf, out, log->zip_size);
+    /* Uncompress tboot log */
+    if (log->zip_size > 0) LZ_Uncompress(log_buf, out, log->zip_size);
     /* log is too big for single printk(), so break it up */
     /* print out the uncompressed log */
     for ( unsigned int curr_pos = 0; curr_pos < 32*1024; 
